parquet/src/main/java/org/apache/iceberg/parquet/MessageTypeToType.java

```diff
@@ -56,36 +60,38 @@
 
   @Override
   public Type message(MessageType message, List<Type> fields) {
-    return struct(message, fields);
+    Type struct = struct(message, fields);
+    return struct != null ? struct : Types.StructType.of(Lists.newArrayList());
   }
 
   @Override
   public Type struct(GroupType struct, List<Type> fieldTypes) {
-    if (struct == root) {
-      nextId = 1; // use the reserved IDs for the root struct
-    }
-
     List<org.apache.parquet.schema.Type> parquetFields = struct.getFields();
-    List<Types.NestedField> fields = Lists.newArrayListWithExpectedSize(fieldTypes.size());
+    List<Types.NestedField> fields = Lists.newArrayList();
```

Comment:
I think it would be fine to continue to use `withExpectedSize` because the size will actually be smaller.

Reply:
Agree